drivers: fuelgauge: Add Onsemi LC709203F driver and refactor fuelgauge sample

DBS06 opened the pull request:
This PR adds the following:
- Add driver for the Onsemi LC709203F fuel gauge
- refactor fuelgauge sample to be more generic

I had to redo the PR #89000, because in my original PR I messed up the commits due to GitHub's merge conflict solver.

DBS06 committed:
drivers: fuelgauge: format fuel_gauge.h

format fuel_gauge.h according to the clang-format settings

Signed-off-by: Philipp Steiner <philipp.steiner1987@gmail.com>

DBS06 committed:
tests: fuelgauge: Add Onsemi LC709203F driver

Add test for the Onsemi LC709203F fuel gauge

Signed-off-by: Philipp Steiner <philipp.steiner1987@gmail.com>

DBS06 committed:
drivers: fuelgauge: Add additional properties

Adds properties for current direction, charge and voltage alarms

Signed-off-by: Philipp Steiner <philipp.steiner1987@gmail.com>

DBS06 committed:
drivers: fuelgauge: Add Onsemi LC709203F driver

Add driver for the Onsemi LC709203F fuel gauge

Signed-off-by: Philipp Steiner <philipp.steiner1987@gmail.com>

DBS06 committed:
boards: adafruit: esp32s2_feather: update documentation

update because of LC709203F driver was added

Signed-off-by: Philipp Steiner <philipp.steiner1987@gmail.com>

DBS06 commented:
@kartben @teburd As I mentioned In the PR #89000 I had to reopen the PR due to GitHub's merge conflict solver, which messed up my branch commits and git log.

kartben commented on tests/drivers/fuel_gauge/lc709203f/src/test_lc709203f.c:
```diff
@@ -0,0 +1,195 @@
+		      (vals[5].current_direction == 0x0001) ||
+		      (vals[5].current_direction == 0xFFFF)));
+	zassert_equal(vals[6].state_of_charge_alarm, 0x0008);
+	zassert_equal(vals[7].state_of_charge_alarm, 0x0000);
```
typo -- but prob means there's a bug somewhere if it wasn't caught when running the tests?

```suggestion
	zassert_equal(vals[7]. low_voltage_alarm, 0x0000);
```

kartben commented on samples/fuel_gauge/README.rst:
````diff
@@ -0,0 +1,85 @@
+Sample output
+*************
+
+```
````
this is markdown, we do ReST :)
it should be 
```
.. code-block::
   bla
   bla
```

DBS06 commented on drivers/fuel_gauge/lc709203f/emul_lc709203f.c:
```diff
@@ -0,0 +1,271 @@
+	switch (reg) {
+	case LC709203F_REG_RSOC:
+		data->regs[LC709203F_REG_RSOC] = value;
+		data->regs[LC709203F_REG_CELL_ITE] = value / 10;
```
Good catch!

kartben replied:
Full disclosure, it's Gemini 2.5 who caught it :) but I agree!

DBS06 commented on samples/fuel_gauge/README.rst:
Good to know, I have been wondering the whole time why the fuel_gauge sample is not under samples/drivers.
For sure I will fix that.

> what's more, the sample doesn't show up [here](https://builds.zephyrproject.io/zephyr/pr/88059/docs/samples/index.html) so something's off

I think you parsed the link to the wrong PR-Build, see [here](https://builds.zephyrproject.io/zephyr/pr/90310/docs/samples/index.html). There it will be shown.

DBS06 commented:
@kartben I already added your requested changes locally, but I am waiting for the review from @aaronemassey, he announced on discord that he is currently reviewing my PR and it should be available today and when he is done I will push everything at once, and besides that I don't want to interfere his current review.

DBS06 committed:
samples: fuelgauge: unify fuel gauge sample

Replace IC specific fuel gauge example with a generic fuel gauge example
and move the fuel gauge sample to sample/drivers folder

Signed-off-by: Philipp Steiner <philipp.steiner1987@gmail.com>
